### PR TITLE
Cleanup prototype comments in nullable tests (Part 2)

### PR DESCRIPTION
**Pull request by AlekseyTs**

### CODE REVIEW

File: src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs [hunk 1]
```diff
@@ -21183,8 +21190,7 @@ void Test2<T2>(T2 x2) where T2 : class, new()
                 );
         }
 
-        // PROTOTYPE(NullableReferenceTypes): Report warnings
-        // if `where T : new()` implies T is non-nullable.
+        // `where T : new()` does not imply T is non-nullable.
```
Comment (333fred) [inline src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs]: Update the design doc?

Comment (AlekseyTs) [reply src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs]: > Update the design doc?

I do not believe there is anything to update

---
In reply to: [218171343](https://github.com/dotnet/roslyn/pull/29947#discussion_r218171343) [](ancestors = 218171343)

File: src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs [hunk 2]
```diff
@@ -30876,13 +30886,19 @@ static void F(C<object?> x, C<object> y)
                 // (8,13): warning CS8619: Nullability of reference types in value of type 'C<object?>' doesn't match target type 'I<object>'.
                 //         a = x;
                 Diagnostic(ErrorCode.WRN_NullabilityMismatchInAssignment, "x").WithArguments("C<object?>", "I<object>").WithLocation(8, 13),
+                // (9,13): warning CS8619: Nullability of reference types in value of type 'C<object?>' doesn't match target type 'I<object>'.
+                //         a = x!;
+                Diagnostic(ErrorCode.WRN_NullabilityMismatchInAssignment, "x!").WithArguments("C<object?>", "I<object>").WithLocation(9, 13),
```
Comment (333fred) [inline src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs]: More general curiosity: how would one silence this warning, if not by the `!`?

File: src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs (hunk 1 above)
Comment (333fred) [reply src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs]: A note here to document would be useful: [https://github.com/dotnet/roslyn/blob/features%2FNullableReferenceTypes/docs/features/nullable-reference-types.md#type-parameters](https://github.com/dotnet/roslyn/blob/features%2FNullableReferenceTypes/docs/features/nullable-reference-types.md#type-parameters)

---
In reply to: [218172264](https://github.com/dotnet/roslyn/pull/29947#discussion_r218172264) [](ancestors = 218172264,218171343)

Comment (AlekseyTs) [reply src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs]: > A note here to document would be useful: [https://github.com/dotnet/roslyn/blob/features%2FNullableReferenceTypes/docs/features/nullable-reference-types.md#type-parameters](https://github.com/dotnet/roslyn/blob/features%2FNullableReferenceTypes/docs/features/nullable-reference-types.md#type-parameters)

I don't think obvious things and every random questions need to be noted

---
In reply to: [218172966](https://github.com/dotnet/roslyn/pull/29947#discussion_r218172966) [](ancestors = 218172966,218172264,218171343)

File: src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs (hunk 2 above)
Comment (AlekseyTs) [reply src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs]: > More general curiosity: how would one silence this warning, if not by the !?

I'll open an issue for this question

---
In reply to: [218172518](https://github.com/dotnet/roslyn/pull/29947#discussion_r218172518) [](ancestors = 218172518)

Comment (jcouv) [reply src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs]: I think [https://github.com/dotnet/roslyn/issues/29642](https://github.com/dotnet/roslyn/issues/29642) tracks this already

---
In reply to: [218174205](https://github.com/dotnet/roslyn/pull/29947#discussion_r218174205) [](ancestors = 218174205,218172518)

Comment (AlekseyTs) [reply src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs]: > I think [https://github.com/dotnet/roslyn/issues/29642](https://github.com/dotnet/roslyn/issues/29642) tracks this already

Agree 

---
In reply to: [218209547](https://github.com/dotnet/roslyn/pull/29947#discussion_r218209547) [](ancestors = 218209547,218174205,218172518)

**Pull request merged**